src/axom/sina/interface/sina_fortran_interface.f.in

```diff
@@ -1,152 +1,332 @@
+    character(*), intent(in) :: name
+    character(*), intent(in) :: curve
+    real, intent(in) :: values(n)
+    integer, intent(in) :: n
+    logical, intent(in) :: independent
+    character(*), optional, intent(in) :: rec
+    
+    integer(c_int) :: indep_int
+    character(:), allocatable :: name_c, curve_c, rec_c
+    
+    if (independent) then
+      indep_int = 1
+    else
+      indep_int = 0
+    endif
+    
+    name_c = make_cstring(name)
+    curve_c = make_cstring(curve)
+    if (present(rec)) then
+      rec_c = make_cstring(rec)
+    else
+      rec_c = make_cstring('')
+    endif
+    
+    call sina_add_curve_float_c(name_c, curve_c, values, n, indep_int, rec_c)
+  end subroutine sina_add_curve_float
+
+  subroutine sina_add_curve_int(name, curve, values, n, independent, rec)
+    character(*), intent(in) :: name
+    character(*), intent(in) :: curve
+    integer(KIND=4), intent(in) :: values(n)
+    integer, intent(in) :: n
+    logical, intent(in) :: independent
+    character(*), optional, intent(in) :: rec
+    
+    integer(c_int) :: indep_int
+    character(:), allocatable :: name_c, curve_c, rec_c
+    
+    if (independent) then
+      indep_int = 1
+    else
+      indep_int = 0
+    endif
+    
+    name_c = make_cstring(name)
+    curve_c = make_cstring(curve)
+    if (present(rec)) then
+      rec_c = make_cstring(rec)
+    else
+      rec_c = make_cstring('')
+    endif
+    
+    call sina_add_curve_int_c(name_c, curve_c, values, n, indep_int, rec_c)
+  end subroutine sina_add_curve_int
+
+  subroutine sina_add_curve_long(name, curve, values, n, independent, rec)
+    character(*), intent(in) :: name
+    character(*), intent(in) :: curve
+    integer(KIND=8), intent(in) :: values(n)
+    integer, intent(in) :: n
+    logical, intent(in) :: independent
+    character(*), optional, intent(in) :: rec
+    
+    integer(c_int) :: indep_int
+    character(:), allocatable :: name_c, curve_c, rec_c
+    
+    if (independent) then
+      indep_int = 1
+    else
+      indep_int = 0
+    endif
+    
+    name_c = make_cstring(name)
+    curve_c = make_cstring(curve)
+    if (present(rec)) then
+      rec_c = make_cstring(rec)
+    else
+      rec_c = make_cstring('')
+    endif
+    
+    call sina_add_curve_long_c(name_c, curve_c, values, n, indep_int, rec_c)
+  end subroutine sina_add_curve_long
 
+  ! Keep your existing make_cstring function here
+  function make_cstring(fstr) result(cstr)
+    character(*), intent(in) :: fstr
+    character(len=len_trim(fstr)+1, kind=c_char) :: cstr
+    if (len_trim(fstr) > 0) then
+      cstr(1:len_trim(fstr)) = fstr(1:len_trim(fstr))
+    endif
+    cstr(len_trim(fstr)+1:len_trim(fstr)+1) = c_null_char
+  end function make_cstring
+
+
+end module sina_functions
 module sina_hdf5_config
   implicit none
   logical, parameter :: use_hdf5 = @AXOM_USE_HDF5_FORTRAN@
-end module sina_hdf5_config
+end module sina_hdf5_config
```

Comment:
Is this file missing a trailing newline?

Reply:
fixed